File: services/user-service/src/main/java/com/visor/school/userservice/dto/UserResponse.java
package com.visor.school.userservice.dto;

import java.util.Set;
import java.util.UUID;

import com.visor.school.userservice.model.User;
import com.visor.school.userservice.model.UserRole;

public record UserResponse(
    UUID id,
    String email,
    String firstName,
    String lastName,
    Set<UserRole> roles,
    boolean emailVerified,
    String accountStatus
) {
    public static UserResponse from(User user) {
        return new UserResponse(
            user.getId(),
            user.getEmail(),
            user.getFirstName(),
            user.getLastName(),
            user.getRoles(),
            user.isEmailVerified(),
            user.getAccountStatus() != null ? user.getAccountStatus().name() : null
        );
    }
}

File: services/user-service/src/main/kotlin/com/visor/school/userservice/dto/UserResponse.kt
package com.visor.school.userservice.dto

import com.visor.school.userservice.model.User
import com.visor.school.userservice.model.UserRole
import java.util.UUID

// Response DTOs
data class UserResponse(
    val id: UUID,
    val email: String,
    val firstName: String,
    val lastName: String,
    val roles: Set<UserRole>,
    val emailVerified: Boolean,
    val accountStatus: String
) {
    companion object {
        fun from(user: User): UserResponse {
            return UserResponse(
                id = user.id!!,
                email = user.email,
                firstName = user.firstName,
                lastName = user.lastName,
                roles = user.roles.toSet(),
                emailVerified = user.emailVerified,
                accountStatus = user.accountStatus.name
            )
        }
    }
}